src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,17 @@
 
     @Override
     public User register(User user) {
+        if (user == null) {
+            throw new InvalidDataException("invalid data");
+        } else if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidDataException("problem with login");
+        } else if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidDataException("problem with password");
+        } else if (user.getAge() < 18) {
+            throw new InvalidDataException("problem with age");
+        } else if (storageDao.get(user.getLogin()) == null) {
+            return storageDao.add(user);
+        }
         return null;
```

Comment:
Returning `null` here might not be the best approach. If the user already exists, you might want to throw an exception or return a specific result indicating the user is already registered. Consider revising this logic to handle the case where the user already exists in a more informative way.